### PR TITLE
[MKLDNN] Pass: Fuse Conv + Bias

### PR DESCRIPTION
**Pull request by Sand3r-**
Fuses Convolution with Elementwise used as bias using Pass mechanism. Applying this fuse at CRNN-CTC when python transpiler and mkldnn was used, **doubled** the number of FPS #12669.

### CODE REVIEW

File: paddle/fluid/framework/ir/conv_bias_mkldnn_fuse_pass.cc [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+#include <vector>
+#include "paddle/fluid/platform/enforce.h"
+namespace paddle {
+namespace framework {
+namespace ir {
+std::unique_ptr<ir::Graph> ConvBiasFusePass::ApplyImpl(
+    std::unique_ptr<ir::Graph> graph) const {
+  PADDLE_ENFORCE(graph.get());
+  FusePassBase::Init("conv_bias_mkldnn_fuse", graph.get());
+  std::unordered_set<Node*> nodes2delete;
```
Comment (sfraczek) [inline paddle/fluid/framework/ir/conv_bias_mkldnn_fuse_pass.cc]: This looks like unused variable

**Review by sfraczek — CHANGES_REQUESTED** [review]
You may want to take a look how I have skipped deletion of convolution :) 
https://github.com/PaddlePaddle/Paddle/pull/13552

**Comment by Sand3r-** [comment]
@sfraczek Unfortunately I cannot reuse the conv op, since setting the bias buffer to inexistant tensor causes the biases to have no assigned operator. As a result aren't linked to anything and remain floating "in air".

**Review by sfraczek — DISMISSED** [review]
LGTM!

**Comment by Sand3r-** [comment]
@luotao1 Is there anything I could do to improve this PR?

File: paddle/fluid/framework/naive_executor.cc [hunk 2]
```diff
@@ -146,5 +146,22 @@ void NaiveExecutor::CleanFeedFetchOps() {
   ops_.swap(ops);
 }
 
+void NaiveExecutor::EnableMKLDNN(const ProgramDesc &program) {
```
Comment (luotao1) [inline paddle/fluid/framework/naive_executor.cc]: #13727 has the duplicated code. You can merge with the latest code.

**Review by luotao1 — COMMENTED** [review]
> Is there anything I could do to improve this PR

You can merge the latest code to see whether `conv+bias` fuse pass is successful in `test_analyzer_resnet50`

**Comment by Sand3r-** [comment]
@luotao1 I have enabled MKLDNN in the `test_analyzer_resnet50`, but it fails due to inaccuracy reasons discussed in: #13646 

Can I disable it for now and have this PR merged and then ask @sfraczek to enable it in his own PR of #13486 which fixes the aforementioned problem?

This conv+bias pass is already tested in `test_analyzer_ocr` and that test passes successfully. Testing it with resnet50 doesn't make much sense without conv+bn fuse, since convolutions in resnet50 don't have their own biases anyway.

**Comment by luotao1** [comment]
>  Testing it with resnet50 doesn't make much sense without conv+bn fuse, since convolutions in resnet50 don't have their own biases anyway.

The model in `test_analyzer_resnet50` is `conv+elementwise_add+batch_norm`. After conv+bn fuse, it still has `conv+elementwise_add`. Thus, your PR will be used in `test_analyzer_resnet50` case.

**Comment by luotao1** [comment]
> Can I disable it for now and have this PR merged

Yes, you can disable MKLDNN in `test_analyzer_resnet50` and have this PR merged at first.

**Review by luotao1 — APPROVED** [review]
LGTM

**Pull request merged**